modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js

```diff
@@ -67,183 +87,170 @@ const InvitesPage = ({
-				if (partnerManagerIndex === -1) {
-					filterRoles = [
-						...filterRoles,
-						ROLE_TYPES.partnerManager.key,
-					];
-				}
+				setAccountMemberRole(accountMember);
+
+				setFieldValue(
+					'invites[0].role',
+					maxRequestors < 2
+						? accountMember
+						: roles?.find(
+								({name}) =>
+									name === ROLE_TYPES.requestor.name ||
+									name === ROLE_TYPES.admin.name
+						  )
+				);
 
-				if (partnerMemberIndex === -1) {
-					filterRoles = [
-						...filterRoles,
-						ROLE_TYPES.partnerMember.key,
-					];
+				for (let i = 1; i < INITIAL_INVITES_COUNT; i++) {
+					setFieldValue(`invites[${i}].role`, accountMember);
 				}
-			}
-
-			setFieldValue(
-				'invites[0].roleId',
-				maxRequestors === 1
-					? ROLE_TYPES.member.key
-					: filterRoles.find(
-							(role) => role === ROLE_TYPES.requestor.key
-					  ) || ROLE_TYPES.admin.key
-			);
 
-			const mapRolesName = filterRoles.sort().map((role) => {
-				const roleProperty = Object.values(ROLE_TYPES).find(
-					({key}) => key === role
+				setAccountRoles(roles);
+				setAccountRolesOptions(
+					roles?.map((role) => ({
+						disabled: false,
+						label: role.name,
+						value: role.id,
+					}))
 				);
+			}
+		};
 
-				return {
-					disabled: false,
-					label: roleProperty?.name || role,
-					value: roleProperty?.key || role,
-				};
-			});
-			setAccountRoles(mapRolesName);
-		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [project, rolesData]);
+		getRoles();
+	}, [maxRequestors, project, projectHasSLAGoldPlatinum, setFieldValue]);
 
 	useEffect(() => {
-		if (values && rolesData) {
-			const totalAdmins = values.invites.reduce(
+		if (values && accountRoles?.length) {
+			const totalAdmins = values.invites?.reduce(
 				(totalInvites, currentInvite) => {
 					if (
-						currentInvite.roleId === ROLE_TYPES.requestor.key ||
-						currentInvite.roleId === ROLE_TYPES.admin.key
+						currentInvite.role.name === ROLE_TYPES.requestor.name ||
+						currentInvite.role.name === ROLE_TYPES.admin.name
 					) {
-						const total = totalInvites + 1;
-
-						return total;
+						return ++totalInvites;
 					}
 
 					return totalInvites;
 				},
 				1
 			);
+
 			const remainingAdmins = maxRequestors - totalAdmins;
 
-			disableAdminOptions(remainingAdmins === 0);
+			if (remainingAdmins < 1) {
+				setAccountRolesOptions((previousAccountRoles) =>
+					previousAccountRoles.map((previousAccountRole) => ({
+						...previousAccountRole,
+						disabled:
+							previousAccountRole.label ===
+								ROLE_TYPES.requestor.name ||
+							previousAccountRole.label === ROLE_TYPES.admin.name,
+					}))
+				);
+			}
 
 			setAvailableAdminsRoles(remainingAdmins);
 		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [values, project, rolesData]);
+	}, [values, project, maxRequestors, accountRoles]);
 
 	useEffect(() => {
+		const filledEmails =
+			values?.invites?.filter(({email}) => email)?.length || 0;
```

Comment:
Do we need this {} around the email here?

Reply:
yeah. The invite is a object with email and role properties. Using the {} it's for use only the email value